src/python/WMCore/MicroService/MSPileup/DataStructs/MSPileupObj.py

```diff
@@ -0,0 +1,139 @@
+    "lastUpdateTime": int, seconds since epoch in GMT timezone
+    "expectedRSEs": ["Disk1", "Disk2", etc],  # these values need to be validated against Rucio
+    "currentRSEs": ["Disk1", "Disk3"],  # values provided by the MS itself
+    "fullReplicas": integer,  # total number of replicas to keep on Disk
+    "campaignList": [ "name", ... ] # list of workflow campaigns using this pileup
+    "containerFraction": real number with the container fraction to be distributed (TBFD)
+    "replicationGrouping": string with a constant value (DATASET or ALL, to be in sync with Rucio)
+    "activatedOn": int, seconds since epoch in GMT timezone
+    "deactivatedOn": int, seconds since epoch in GMT timezone
+    "active": boolean,
+    "pileupSizeGB": integer, current size of the pileup in GB (integer)
+    "rulesList: list of strings (rules) used to lock the pileup id
+}
+
+The data flow should be done via list of objects, e.g.
+[<pileup object 1>, <pileup object 2>, ..., <pileup object n>]
+"""
+
+# system modules
+import time
+import json
+
+# WMCore modules
+from WMCore.MicroService.Tools.Common import getMSLogger
+
+
+def gmtimeSeconds():
+    """
+    Return GMT time in seconds
+    """
+    return int(time.mktime(time.gmtime()))
+
+
+class MSPileupObj(object):
+    """
+    MSPileupObj defines MSPileup data stucture
+    """
+    def __init__(self, pdict, verbose=None, logger=None):
+        self.logger = getMSLogger(verbose, logger)
+
+        self.pileupName = pdict.get('pileupName', '')
+        self.pileupType = pdict.get('pileupType', 'classic')
+        self.insertTime = pdict.get('insertTime', gmtimeSeconds())
+        self.lastUpdateTime = pdict.get('lastUpdateTime', gmtimeSeconds())
+        self.expectedRSEs = pdict.get('expectedRSEs', [])
+        self.fullReplicas = pdict.get('fullReplicas', 0)
+        self.campaigns = pdict.get('campaigns', [])
+        self.containerFraction = pdict.get('containerFraction', 0.0)
+        self.replicationGrouping = pdict.get('replicationGrouping', "")
+        self.activatedOn = pdict.get('activatedOn', gmtimeSeconds())
+        self.deactivatedOn = pdict.get('deactivatedOn', gmtimeSeconds())
+        self.active = pdict.get('active', False)
+        self.pileupSize = pdict.get('pileupSize', 0)
+        self.ruleList = pdict.get('ruleList', [])
+
+        self.data = {
+            'pileupName': self.pileupName,
+            'pileupType': self.pileupType,
+            'insertTime': self.insertTime,
+            'lastUpdateTime': self.lastUpdateTime,
+            'expectedRSEs': self.expectedRSEs,
+            'fullReplicas': self.fullReplicas,
+            'campaigns': self.campaigns,
+            'containerFraction': self.containerFraction,
+            'replicationGrouping': self.replicationGrouping,
+            'activatedOn': self.activatedOn,
+            'deactivatedOn': self.deactivatedOn,
+            'active': self.active,
+            'pileupSize': self.pileupSize,
```

Comment:
I see you added the two optional parameters. This was a good call, I really think they will be useful as it gets more mature.